src/constructs/aws/DatabaseSql.ts

```diff
@@ -0,0 +1,161 @@
+
+export class DatabaseSql extends AwsConstruct {
+    public static type = "database/sql";
+    public static schema = SCHEMA;
+
+    private readonly dbInstance: DatabaseInstance;
+    private readonly dbHostOutput: CfnOutput;
+    private readonly username: string;
+    private passwordSecretName: string | undefined;
+
+    constructor(
+        scope: CdkConstruct,
+        private readonly id: string,
+        private readonly configuration: Configuration,
+        private readonly provider: AwsProvider
+    ) {
+        super(scope, id);
+
+        const vpc = provider.enableVpc();
+
+        this.username = this.configuration.username ?? "admin";
+
+        this.dbInstance = new DatabaseInstance(this, "Instance", {
+            // https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseInstance.html#construct-props
+            instanceIdentifier: configuration.name ?? `${this.provider.stackName}-${id}`,
+            databaseName: this.safeDbName(configuration.name ?? `${this.provider.stackName}-${id}`),
+            engine: this.getEngineVersion(),
+            instanceType: new InstanceType(configuration.instanceType ?? "t3.micro"),
+            credentials: this.credentials(),
+            vpc,
+            // Put the instance in the private subnet
+            vpcSubnets: {
+                subnetType: SubnetType.PRIVATE,
+            },
+            // We go with 20 (the minimum) instead of 100 by default, because it is much cheaper
+            allocatedStorage: configuration.storageSize ?? 20,
+            // 7 days backups instead of the default 1
+            backupRetention: Duration.days(7),
+            cloudwatchLogsExports: this.logsToEnable(),
+        });
+
+        this.dbHostOutput = new CfnOutput(this, "DbHost", {
+            value: this.dbInstance.instanceEndpoint.hostname,
+        });
+    }
+
+    private credentials() {
+        const password = this.configuration.password ?? "";
+        if (password === "") {
+            // If no password is defined, a random one will be generated and stored in Secrets Manager
+            this.passwordSecretName = `${this.provider.stackName}/${this.id}/password`;
+
+            return Credentials.fromGeneratedSecret(this.username, {
+                secretName: this.passwordSecretName,
+            });
+        }
+
+        return Credentials.fromPassword(this.username, SecretValue.plainText(password));
+    }
+
+    commands(): Record<string, () => void | Promise<void>> {
+        return {};
+    }
+
+    outputs(): Record<string, () => Promise<string | undefined>> {
+        return {
+            host: () => this.provider.getStackOutput(this.dbHostOutput),
+        };
+    }
+
+    variables(): Record<string, unknown> {
+        return {
+            host: this.dbInstance.instanceEndpoint.hostname,
+            port: this.dbInstance.instanceEndpoint.port,
+            username: this.username,
+            passwordSecret: this.passwordSecretName,
```

Comment:
I think I misunderstood what you did here.

The idea was to fetch the secret from the application code, not via serverless framework integration with secret manager.

Wouldn't that need extra permissions?

This also comes with a performance penalty because it has to be fetched everytime.

Reply:
Ah yes, yes and yes!

That's why there are 2 approaches:

1. deploy with the password in clear:
  - that's easier, simpler
  - but also less secure: the password is in clear in the Lambda console
  - and finally it prevents automatic rotation of DB passwords (because the Lambda function would need to be redeployed to get the new password)
  - free, with either environment variables, SSM parameter store, Serverless Dashboard secrets, etc.
2. deploy with the name of the secret: the application needs to retrieve the secret
  - more secure, as the password is decrypted at runtime
  - negative performance impact
  - benefits from password rotation
  - more expensive, as it requires Secrets Manager which has a fixed cost

That's why I think we need to provide support for both options.